### PR TITLE
mkmf.rb: use kwargs

### PR DESCRIPTION
**Pull request by nobu**

### CODE REVIEW

File: lib/mkmf.rb [hunk 1]
```diff
@@ -679,19 +679,19 @@ def with_cppflags(flags)
   end
 
   # :nodoc:
-  def try_cppflags(flags, opts = {})
-    try_header(MAIN_DOES_NOTHING, flags, {:werror => true}.update(opts))
+  def try_cppflags(flags, werror: true, **opts)
+    try_header(MAIN_DOES_NOTHING, flags, werror: werror, **opts)
```
Comment (sorah) [inline lib/mkmf.rb:683]:
```suggestion
    try_header(MAIN_DOES_NOTHING, flags, werror:, **opts)
```

File: lib/mkmf.rb [hunk 2]
```diff
@@ -710,8 +710,8 @@ def with_cflags(flags)
   end
 
   # :nodoc:
-  def try_cflags(flags, opts = {})
-    try_compile(MAIN_DOES_NOTHING, flags, {:werror => true}.update(opts))
+  def try_cflags(flags, werror: true, **opts)
+    try_compile(MAIN_DOES_NOTHING, flags, werror: werror, **opts)
```
Comment (sorah) [inline lib/mkmf.rb:714]:
```suggestion
    try_compile(MAIN_DOES_NOTHING, flags, werror:, **opts)
```

File: lib/mkmf.rb [hunk 3]
```diff
@@ -726,9 +726,8 @@ def with_ldflags(flags)
   end
 
   # :nodoc:
-  def try_ldflags(flags, opts = {})
-    opts = {:werror => true}.update(opts) if $mswin
-    try_link(MAIN_DOES_NOTHING, flags, opts)
+  def try_ldflags(flags, werror: $mswin, **opts)
+    try_link(MAIN_DOES_NOTHING, flags, werror: werror, **opts)
```
Comment (sorah) [inline lib/mkmf.rb:730]:
```suggestion
    try_link(MAIN_DOES_NOTHING, flags, werror:, **opts)
```

**Review by sorah — APPROVED** [review]

File: lib/mkmf.rb (hunk 1 above)
Comment (nobu) [reply lib/mkmf.rb:683]: mkmf.rb will be used also by BASERUBY, which may not support forwarding keyword arguments.

**Pull request merged**